src/app/pages/view/view-records/view-records.component.ts

```diff
@@ -78,22 +79,23 @@ export class ViewRecordsComponent implements OnInit {
         this.records = null;
 
         let datasetParams: any = {};
-        let recordParams: any = {};
+
+        this.recordParams = {};
```

Comment:
Do this when you declare the field, that way it will infer the type (TypeScript approves).